solidity/random-beacon/README.adoc

```diff
@@ -177,103 +166,122 @@ When the hard timeout is reached, anyone can notify about this fact by calling
 operator that was already marked as ineligible for rewards. This approach does
 not increase the gas cost of a happy path and leaves some freedom to group
-members. They may mark as ineligible operators who turned off their nodes,
+members. They can mark as ineligible operators who turned off their nodes,
 operators whose nodes never participate in signing because they are
 misconfigured, or operators who notoriously miss their turn in submitting relay
-entry.
+entries.
+
+=== Rewards
+
+T rewards are allocated to all operators registered in the beacon sortition
+pool, excluding operators who were marked as ineligible for rewards as a result
+of <<inactivity, inactivity>>. Rewards are allocated proportionally to the
+operator's weight in the pool. 
+
+[[transaction-incentives]]
+=== Transaction Incentives
+
+There are two types of transactions: <<operator-only,Operator-Only>>,
+<<public-knowledge,Public-Knowledge>>, and <<punishment,Punishment>>.
+
+[[operator-only]]
+==== Operator-Only
+Operator-Only transactions are where only the operators have access to the
+information required to assemble the transaction with the right input
+parameters.
+
+In order to avoid all operators racing to submit the transaction at the same
+time, we have an off-chain informal agreement to submit based on the operator's
+position in the group (can use the hash of the group's pubkey).
+
+If the designated operator does not submit their transaction before a timeout
+expires, the duty moves to the next operator and the group can sign a
+transaction to mark that operator as inactive. Since there is no slashing
+reward, and since this transaction can only be submitted by an operator, this
+transaction is also Operator-Only.
+
+In order to compensate the operator for posting the transaction, the gas spent
+will be reimbursed by a DAO-funded eth pool in the same transaction (see
+`Reimbursable` and `ReimbursementPool` smart contracts).
+
+Operator-only transactions are `submitDkgResult`, `submitRelayEntry`,
+`notifyOperatorInactivity`, and `approveDkgResult` for a certain number of
+blocks, before a timeout for the original DKG result submitter to call this
+function elapses.
+
+[[public-knowledge]]
+==== Public-Knowledge
+Public-Knowledge transactions are where anyone has access to the information
+required to assemble the transaction and the transaction does not lead to
+punishment.
+
+In order to prevent wasting gas on racing to submit, such transactions need to
+be executed rarely, and off-chain clients should follow the informal agreement
+about the submission order.
+
+To compensate these transactions, whoever posts them will have the gas spent
+reimbursed by a DAO-funded eth pool in the same transaction.
+
+The only public knowledge transaction is `notifyDkgTimeout`.
+
+`approveDkgResult` turns into a public knowledge transaction in case the
+original submitter has not approved the result before the timeout.
+
+[[punishment]]
+==== Punishment
```

Comment:
Not sure where it would be the best place to put it, but we could mention somewhere what happens to the slashed amount - part goes as a reward, what happens to the other part?

Reply:
It gets burned. I think it's a logic of a staking contract and we shouldn't talk about it. It is getting quite complicated when we go into details.

Reply:
OK, let's leave it as it is.